packages/stream_video/lib/src/core/client_state.dart

```diff
@@ -58,22 +73,54 @@ class MutableClientState implements ClientState {
   User get currentUser => user.value;
 
   Future<void> clear() async {
-    activeCall.value = null;
+    activeCalls.value = [];
     outgoingCall.value = null;
     connection.value = ConnectionState.disconnected(user.value.id);
   }
 
-  Call? getActiveCall() => activeCall.valueOrNull;
+  Call? getActiveCall() {
+    if (options.allowMultipleActiveCalls) {
+      throw Exception(
+        'Multiple active calls are enabled, use getActiveCalls() instead',
+      );
+    }
+
+    return activeCalls.value.firstOrNull;
+  }
+
+  List<Call> getActiveCalls() => activeCalls.value;
   Call? getOutgoingCall() => outgoingCall.valueOrNull;
 
   @override
   Future<void> setActiveCall(Call? call) async {
-    final currentlyActiveCall = activeCall.valueOrNull;
-    if (currentlyActiveCall != null && call != null) {
-      await currentlyActiveCall.leave();
+    if (!options.allowMultipleActiveCalls) {
+      final currentlyActiveCall = activeCalls.value.firstOrNull;
+      if (currentlyActiveCall != null) {
+        await currentlyActiveCall.leave(reason: DisconnectReason.replaced());
+      }
+
+      activeCall.value = call;
+    }
+
+    if (call == null) {
+      activeCalls.value = [];
+    } else if (!options.allowMultipleActiveCalls) {
+      activeCalls.value = [call];
+    } else {
+      activeCalls.value = [...activeCalls.value, call];
+    }
+  }
```

Comment:
I think we should only do this in case 1 call is allowed:
```dart
    if (call == null) {
      activeCalls.value = [];
    }
```
So I would move that up to the `if (!options.allowMultipleActiveCalls)` and merge it together. You have 3 cases now only for `if (!options.allowMultipleActiveCalls)`.

```suggestion
    if (!options.allowMultipleActiveCalls) {
      final currentlyActiveCall = activeCalls.value.firstOrNull;
      if (currentlyActiveCall != null) {
        await currentlyActiveCall.leave(reason: DisconnectReason.replaced());
      }

      activeCall.value = call;
      activeCalls.value = call == null ? [] : [call];
    } else if (call != null) {
      activeCalls.value = [...activeCalls.value, call];
    }
  }
```